torch/serialization.py

```diff
@@ -394,29 +329,32 @@ def _validate_privateuse1_device(location, backend_name):
                                f'{device_index} but torch.{backend_name}.device_count() is {device_count}. '
                                'Please use torch.load with map_location to map your storages '
                                'to an existing device.')
-    return device_index
+    return device
+
+
+def validate_cuda_device(location):
+    return _validate_device(location, 'cuda').index
 
 
-def _privateuse1_deserialize(obj, location):
-    backend_name = torch._C._get_privateuse1_backend_name()
+def validate_hpu_device(location):
+    return _validate_device(location, 'hpu').index
```

Comment:
Why can't these use the _validate_device function above? Is it to preserve public API?

cc @mikaylagawarecki should we deprecate/remove these public APIs?

Reply:
> Why can't these use the _validate_device function above? Is it to preserve public API?

Yes, that's to preserve public API. Note that if `validate_cuda_device` and `validate_hpu_device` are removed, then BC-Linter notices this and fails. So, I decided to add these simple wrappers. If you believe it's time to break backward compatibility - let me know, I'll remove them. But you will need help me with silencing BC-Linter since I am not familiar with the process of how backward incompatible changes are managed in pytorch.

Reply:
We can do these in a follow up if @mikaylagawarecki wants it.
We can keep it as is for this PR for sure.

Reply:
Ok, appreciate if this can be done in a follow up PR. FYI, inside pytorch code only `validate_cuda_device` is called, here:
https://github.com/pytorch/pytorch/blob/3d20a5dd87e9c45e20dbc89210cfc63e7824b2b9/torch/jit/_serialization.py#L181-L182

I don't see calls to `validate_hpu_device`, though both are declared as public APIs and reported to documentation.